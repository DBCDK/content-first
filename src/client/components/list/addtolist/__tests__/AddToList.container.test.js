import React from 'react';
import {AddToList} from '../AddToList.container';
import {shallow} from 'enzyme';

const createTestElement = id => {
  return {
    pid: 'pid' + id,
    links: {
      cover: '/cover' + id
    },
    description: 'some description' + id
  };
};

const createList = _id => {
  return {
    _id,
    title: 'some list',
    description: 'some description',
    list: [createTestElement(1), createTestElement(2), createTestElement(3)]
  };
};

describe('AddToList', () => {
  test('is empty when list is undefined', () => {
    const tree = shallow(<AddToList allowAdd={true} />);
    expect(tree).toMatchSnapshot();
  });
  test('is empty when allowAdd is false', () => {
    const tree = shallow(<AddToList list={createList(1)} allowAdd={false} />);
    expect(tree).toMatchSnapshot();
  });
  test('render when allowAdd is true', () => {
    const tree = shallow(<AddToList list={createList(1)} allowAdd={true} />);
    expect(tree).toMatchSnapshot();
  });
  test('show error msg when selected books is already in list', () => {
    const list = createList(1);
    const tree = shallow(<AddToList list={list} allowAdd={true} />);
    tree.setState({elementToAdd: list.list[0]});
    expect(tree).toMatchSnapshot();
  });
  test('show add and cancel buttons when selected book is not in list', () => {
    const list = createList(1);
    const tree = shallow(<AddToList list={list} allowAdd={true} />);
    tree.setState({elementToAdd: createTestElement(10)});
    expect(tree).toMatchSnapshot();
  });
  test('invoke requireLogin, when add button is clicked when not logged in', () => {
    const list = createList(1);
    const element = createTestElement(10);
    let requireLoginInvoked = false;
    let addElementInvoked = false;
    const tree = shallow(
      <AddToList
        list={list}
        allowAdd={true}
        isLoggedIn={false}
        requireLogin={() => (requireLoginInvoked = true)}
        addElement={() => (addElementInvoked = true)}
      />
    );
    tree.instance().submit(element);
    expect(requireLoginInvoked).toBe(true);
    expect(addElementInvoked).toBe(false);
  });
  test('invoke addElement, when add button is clicked when logged in', () => {
    const list = createList(1);
    const element = createTestElement(10);
    let requireLoginInvoked = false;
    let addElementInvoked = false;
    const tree = shallow(
      <AddToList
        list={list}
        allowAdd={true}
        isLoggedIn={true}
        requireLogin={() => (requireLoginInvoked = true)}
        addElementToList={() => (addElementInvoked = true)}
      />
    );
    tree.instance().submit(element);
    expect(requireLoginInvoked).toBe(false);
    expect(addElementInvoked).toBe(true);
  });
});
