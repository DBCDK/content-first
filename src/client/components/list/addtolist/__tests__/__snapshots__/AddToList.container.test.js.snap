// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToList is empty when allowAdd is false 1`] = `""`;

exports[`AddToList is empty when list is undefined 1`] = `""`;

exports[`AddToList render when allowAdd is true 1`] = `
<div
  className="addbook position-relative undefined"
>
  <Title
    className="ml-2 ml-sm-0"
    tag="h2"
    type="title4"
    variant="transform-uppercase"
  >
    <T
      component="list"
      name="addBooksToList"
    />
  </Title>
  <div
    className="d-flex flex-row pt-2"
  >
    <ProfileImage
      size="40"
    />
    <Connect(BookSearchSuggester)
      className="ml-3 mr-5"
      list={
        Object {
          "_id": 1,
          "description": "some description",
          "list": Array [
            Object {
              "book": Object {
                "creator": "some creator1",
                "pid": "pid1",
                "title": "some title 1",
              },
              "description": "some description1",
              "links": Object {
                "cover": "/cover1",
              },
            },
            Object {
              "book": Object {
                "creator": "some creator2",
                "pid": "pid2",
                "title": "some title 2",
              },
              "description": "some description2",
              "links": Object {
                "cover": "/cover2",
              },
            },
            Object {
              "book": Object {
                "creator": "some creator3",
                "pid": "pid3",
                "title": "some title 3",
              },
              "description": "some description3",
              "links": Object {
                "cover": "/cover3",
              },
            },
          ],
          "title": "some list",
        }
      }
      onSubmit={[Function]}
    />
  </div>
</div>
`;

exports[`AddToList show add and cancel buttons when selected book is not in list 1`] = `
<div
  className="addbook position-relative undefined"
>
  <Title
    className="ml-2 ml-sm-0"
    tag="h2"
    type="title4"
    variant="transform-uppercase"
  >
    <T
      component="list"
      name="addBooksToList"
    />
  </Title>
  <div
    className="d-flex flex-row pt-2"
  >
    <ProfileImage
      size="40"
    />
    <Connect(BookSearchSuggester)
      className="ml-3 mr-5"
      list={
        Object {
          "_id": 1,
          "description": "some description",
          "list": Array [
            Object {
              "book": Object {
                "creator": "some creator1",
                "pid": "pid1",
                "title": "some title 1",
              },
              "description": "some description1",
              "links": Object {
                "cover": "/cover1",
              },
            },
            Object {
              "book": Object {
                "creator": "some creator2",
                "pid": "pid2",
                "title": "some title 2",
              },
              "description": "some description2",
              "links": Object {
                "cover": "/cover2",
              },
            },
            Object {
              "book": Object {
                "creator": "some creator3",
                "pid": "pid3",
                "title": "some title 3",
              },
              "description": "some description3",
              "links": Object {
                "cover": "/cover3",
              },
            },
          ],
          "title": "some list",
        }
      }
      onSubmit={[Function]}
    />
  </div>
</div>
`;

exports[`AddToList show error msg when selected books is already in list 1`] = `
<div
  className="addbook position-relative undefined"
>
  <Title
    className="ml-2 ml-sm-0"
    tag="h2"
    type="title4"
    variant="transform-uppercase"
  >
    <T
      component="list"
      name="addBooksToList"
    />
  </Title>
  <div
    className="d-flex flex-row pt-2"
  >
    <ProfileImage
      size="40"
    />
    <Connect(BookSearchSuggester)
      className="ml-3 mr-5"
      list={
        Object {
          "_id": 1,
          "description": "some description",
          "list": Array [
            Object {
              "book": Object {
                "creator": "some creator1",
                "pid": "pid1",
                "title": "some title 1",
              },
              "description": "some description1",
              "links": Object {
                "cover": "/cover1",
              },
            },
            Object {
              "book": Object {
                "creator": "some creator2",
                "pid": "pid2",
                "title": "some title 2",
              },
              "description": "some description2",
              "links": Object {
                "cover": "/cover2",
              },
            },
            Object {
              "book": Object {
                "creator": "some creator3",
                "pid": "pid3",
                "title": "some title 3",
              },
              "description": "some description3",
              "links": Object {
                "cover": "/cover3",
              },
            },
          ],
          "title": "some list",
        }
      }
      onSubmit={[Function]}
    />
  </div>
</div>
`;
